sympy/sets/fancysets.py

```diff
@@ -1443,8 +1441,13 @@ def _union(self, other):
             elif self.polar and other.polar:
                 return ComplexRegion(Union(self.sets, other.sets), polar=True)
 
-        if self == S.Complexes:
-            return self
+        # treat a subset of Reals as complex numbers
+        if other.is_subset(S.Reals):
```

Comment:
I don't think converting other to ComplexRegion is necessary, since under the `if other.is_ComplexRegion:` we are only using `other.sets` which just returns set passed in constructor, and this way we don't incur overhead of ComplexRegion's constructor.

If you think this makes logic of what's happening clearer, I'll change it.